testing/repository_client.go

```diff
@@ -0,0 +1,93 @@
+package vtesting
+
+import (
+	"context"
+    "regexp"
+	"github.com/Shopify/voucher/repository"
+)
+const (
+    githubRegex = "^(?:https?|git)(?:://|@)(?P<vcs>[^/:]+)[/:](?P<org>[^/.]+)/(?P<name>[^/.]+)(?:.git)?"
+    commitRegex = "^(?:https?|git)(?:://|@)(?P<vcs>[^/:]+)[/:](?P<org>[^/.]+)/(?P<name>[^/.]+)/(?:commit)/(?P<hash>[^/.]+)"
+)
+
+type repositoryclient struct {
+    repos map[string]Repository
+}
+
+//NewClient creates a new repository client
+func NewClient() (RepoClient, error) {
+	return &repositoryclient{}, nil
+}
+
+//AddRepository adds to repository to the repository client
+func (c *repositoryclient) AddRepository(org repository.Organization, name string, commits map[string]repository.Commit, branches []repository.Branch) {
+    if(c.repos == nil) {
+        c.repos = make(map[string]Repository)
+    }
+    c.repos[org.Name] = Repository{org, name, commits, branches}
+}
+
+//GetCommitInfo creates commit info with provided
+func (c *repositoryclient) GetCommit(ctx context.Context, commitURL string) (repository.Commit, error) {
+    var commit repository.Commit
+    if len(c.repos) == 0 {
+    	if commitURL == "" {
+    		commit = repository.Commit{"https://github.com/org1/project/commit/abcd", nil, "good", false, nil}
+    	} else {
+    		commit = repository.Commit{commitURL, nil, "bad", true, nil}
+    	}
+    } else {
+        re, _ := regexp.Compile(commitRegex)
```

Comment:
Since we're ignoring the error, this might make us have an unusable regex! You can avoid that by using

```suggestion
        re := regexp.MustCompile(commitRegex)
```

`MustCompile` will panic if the regex is malformed, which will save us from unexpected bugs if someone changes the regex constant!

Reply:
Good to know! Done.